tests/testthat/testCombine.R

```diff
@@ -0,0 +1,60 @@
+# testCombine.R
+
+context("test for Combine SNV and CNA files")
+
+# ==== BEGIN SETUP AND PREPARE =================================================
+# set up tempdir and tempfiles for output files and to be deleted after tests
+OLOG <- as.character(getOption("rete.logfile"))   # save original logfile name
+logFileName(fPath = tempdir(), setOption = TRUE)  # make tempdir() the log dir
+# ==== END SETUP AND PREPARE ===================================================
+
+test_that("parameter errors are correctly handled", {
+    # try no parameter input
+    expect_error(combineSNV_CNA(writeLog=FALSE))
+    # try length-zero or NuLL input vector
+    input_vector <- character()
+    expect_error(combineSNV_CNA(input_vector, writeLog=FALSE))
+    # try wrong file format (e.g. text file instead of rds dataframe) in input vector
+    input_vector[1] <- "CNA.txt"
+    expect_error(combineSNV_CNA(input_vector, writeLog=FALSE))
+    # try invalid output fgX file name (e.g. wrong extension)
+    input_vector[1] <- "CNA.rds"
+    expect_error(combineSNV_CNA(input_vector, "out.txt", writeLog=FALSE))
+})
+
+
+test_that("a sane input gives an expected output", {
+    # provide one valid SNV and one valid CNA file with small number of genes as input
+    # TODO where can I find the sample data for testing? inst/extdata folder has maf & txt formats
+    input_vector = c("CNA.rds", "SNV.rds")
+
+    # test if a log file is saved if log is True
+    combineSNV_CNA(input_vector, writeLog=TRUE)
+    expect_true(files.exists(logName))
+    # test if output file fgX is saved under the path
+    combineSNV_CNA(input_vector, writeLog=FALSE)
+    expect_true(files.exists("gx.rds"))
+    # test if specified output filename is saved under the path
+    combineSNV_CNA(input_vector, fgx = "test.rds", writeLog=FALSE)
+    expect_true(files.exists("gx.rds"))
+
+})
+
+
+test_that("combine received corrupted input data file", {
```

Comment:
You don't have to do too much here ... this whole question of data validation needs to be revisited anyway (later) - and since these are compressed files, I don't think they could even be opened if they are truncated.